graphbolt/src/cuda/index_select_impl.cu

```diff
@@ -46,19 +107,41 @@ torch::Tensor UVAIndexSelectImpl_(torch::Tensor input, torch::Tensor index) {
                                       .dtype(input.dtype())
                                       .device(c10::DeviceType::CUDA));
   DType* input_ptr = input.data_ptr<DType>();
-  IdType* index_ptr = index.data_ptr<IdType>();
   DType* ret_ptr = ret.data_ptr<DType>();
+
+  // Sort the index to improve the memory access pattern.
+  torch::Tensor sorted_index, permutation;
+  std::tie(sorted_index, permutation) = torch::sort(index);
```

Comment:
Can we use CUB sort here? We can see the shape of the input, determine the number of bits required to radix sort the input, and use that for the sort. Should be faster. Like:
```
auto N = input.size(0);
int i = 0;
while ((1ll << i) < N)
  i++;
// pass i into cub::radixsort
```

Reply:
> I didn't author the if checks in this code. In my opinion, the threshold should be modified to simply
<= GPU_CACHE_LINE_SIZE. I think aligned is faster otherwise. You can benchmark it for different values to find out which one is better for sure. When there is 1 or less than 1 cache line of feature dimension, then the nonaligned version is more resource-efficient, meaning has the potential to use fewer SMs.

I see. I will only check the size with one GPU_CACHE_LINE_SIZE.

> Can we use CUB sort here?

Yes, we can. But it requires large changes. I will enable it in a later PR.